go-controller/pkg/cni/udn/primary_network.go

```diff
@@ -104,6 +104,29 @@ func (p *UserDefinedPrimaryNetwork) WaitForPrimaryAnnotationFn(namespace string,
 	}
 }
 
+func (p *UserDefinedPrimaryNetwork) Ensure(namespace string, annotations map[string]string, nadName string) error {
+	// non default network is not related to primary UDNs
+	if nadName != types.DefaultNetworkName {
+		return nil
+	}
```

Comment:
nit: shouldn't this belong in the caller ? Or at least in the constructor ? It feels strange to me to have a `UserDefinedPrimaryNetwork` struct instance that is actually not a primary.

Reply:
yeah, but how do I know that if I only look here ? I don't. IIUC, you're actually saying this check is redundant if called only from within a `Found()` block. Right ? 

... then why have it ?

Reply:
So, we need to do this check to break the `GetPodWithAnnotations` as soon as the the default pod network ovn pod annotation is found, and since we are re-using the logic for the unconfigure we do it there too, is not hurting though.